From: Kai Koehne <kai.koehne@qt.io>
Date: Tue, 3 Mar 2020 16:30:11 +0100
Subject: Fix installing qtwayland without qtquick or opengl support

Unconditionally generate qwayland-server-qt-texture-sharing-unstable-v1.h,
even if the header file will not be used. It is listed in syncqt.profile though,
and because of

  private_headers.CONFIG += no_check_exist

in extensions.pri qmake will unconditionally generate an install rule.

Fixes: QTBUG-82569
Change-Id: I5125382f1e4fc61936515e045a7e6a4a3c608451
Reviewed-by: Joerg Bornemann <joerg.bornemann@qt.io>
Origin: upstream, commit:0f0fbf0a70940379a99618d2069721471e3fc631
---
 src/compositor/extensions/extensions.pri | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/src/compositor/extensions/extensions.pri b/src/compositor/extensions/extensions.pri
index 61ab043e..06e3ef9e 100644
--- a/src/compositor/extensions/extensions.pri
+++ b/src/compositor/extensions/extensions.pri
@@ -8,6 +8,7 @@ WAYLANDSERVERSOURCES += \
     ../extensions/touch-extension.xml \
     ../extensions/qt-key-unstable-v1.xml \
     ../extensions/qt-windowmanager.xml \
+    ../extensions/qt-texture-sharing-unstable-v1.xml \
     ../3rdparty/protocol/text-input-unstable-v2.xml \
     ../3rdparty/protocol/viewporter.xml \
     ../3rdparty/protocol/scaler.xml \
@@ -96,9 +97,6 @@ qtHaveModule(quick) {
         extensions/qwaylandxdgshellintegration.cpp \
 
     qtConfig(opengl) {
-        WAYLANDSERVERSOURCES += \
-            ../extensions/qt-texture-sharing-unstable-v1.xml
-
         HEADERS += \
             extensions/qwltexturesharingextension_p.h
 
-- 
